app/scripts/lib/createEvmMethodsToNonEvmAccountReqFilterMiddleware.ts

```diff
@@ -74,7 +75,13 @@ export default function createEvmMethodsToNonEvmAccountReqFilterMiddleware({
     // TODO: Convert this to superstruct schema
     const isWalletRequestPermission =
       req.method === 'wallet_requestPermissions';
-    if (isWalletRequestPermission && req?.params && Array.isArray(req.params)) {
+    if (
```

Comment:
This was caught as type error after upgrading to the stricter types required by `@metamask/json-rpc-engine` and looks like we were not validating input properly here previously.